src/forms/button.test.jsx

```diff
@@ -18,83 +12,62 @@ describe('Button', () => {
 		});
 	});
 
-	it('applies variant classes for each variant prop', () => {
-		const variants = ['primary', 'fullWidth', 'small', 'bordered'];
+	it('has a default state', () => {
+		const button = mount(<Button>Click Me</Button>);
+		expect(
+			button.getDOMNode().attributes.getNamedItem('data-swarm-button').value
+		).toBe('default');
+	});
 
-		variantTest(Button, BUTTON_CLASS, variants);
+	it('has a primary state', () => {
+		const button = mount(<Button primary>Click Me</Button>);
+		expect(
+			button.getDOMNode().attributes.getNamedItem('data-swarm-button').value
+		).toBe('primary');
 	});
 
-	it('executes onClick when clicked', () => {
-		const onClick = jest.fn();
-		const button = shallow(<Button onClick={onClick} />);
+	it('has a neutral state', () => {
+		const button = mount(<Button neutral>Click Me</Button>);
+		expect(
+			button.getDOMNode().attributes.getNamedItem('data-swarm-button').value
+		).toBe('neutral');
+	});
 
-		button.simulate('click');
-		expect(onClick).toHaveBeenCalled();
+	it('has a bordered state', () => {
+		const button = mount(<Button bordered>Click Me</Button>);
+		expect(
+			button.getDOMNode().attributes.getNamedItem('data-swarm-button').value
+		).toBe('bordered');
 	});
 
-	it('does not execute onClick when disabled', () => {
```

Comment:
Should we also have this test? make sure not to call onclick when the button is disabled?

Reply:
This is a test that predates the support being default in react 🦖